File: source/downloads/code/LeetCode/Java/Water-and-Jug-Problem.java
public class Solution {
    public boolean canMeasureWater(int x, int y, int z) {
        if (x+y<z) return false;
        // case: x==0 or y==0
        if (x==z || y==z || x+y==z) return true;
        return z%GCD(x,y)==0;        
    }

    public int GCD(int a, int b){
        while(b!=0){
            int tmp = b;
            b = a%b;
            a = tmp;
        }
        return a;
    }

}

File: downloads/code/LeetCode/Python/Water-and-Jug-Problem.py
class Solution(object):
    def canMeasureWater(self, x, y, z):
        """
        :type x: int
        :type y: int
        :type z: int
        :rtype: bool
        """
        def findGCD(x,y):
            if x == 0:
                return y
            return findGCD(y%x,x)

        if x > y:
            x,y = y,x

        gcd = findGCD(x,y)
        if gcd == 0:
            return z == 0
        return z % gcd == 0 and z <= x+y
